source/Xamarin.Kotlin.StdLib/Transforms/Metadata.xml

```diff
@@ -210,9 +210,105 @@
   <!--
   <remove-node path="/api/package/class[implements[@name='java.util.Collection']]/method[@name='getSize' and count(parameter)=0]" />
   -->
+
   <remove-node path="/api/package/class[@name='UByteArray']/implements[@name='java.util.Collection']" />
   <remove-node path="/api/package/class[@name='UIntArray']/implements[@name='java.util.Collection']" />
   <remove-node path="/api/package/class[@name='UShortArray']/implements[@name='java.util.Collection']" />
   <remove-node path="/api/package/class[@name='ULongArray']/implements[@name='java.util.Collection']" />
 
+  <add-node
+    path="/api/package[@name='kotlin']/class[@name='UShortArray']"
+    >
+    <method 
+      visibility="public" static="false" abstract="false" return="int" name="size"
+      deprecated="not deprecated" final="true" bridge="true" native="false" synchronized="false" 
+      synthetic="false" 
+      >
+    </method>
+  </add-node>
+  <add-node
+    path="/api/package[@name='kotlin']/class[@name='ULongArray']"
+    >
+    <method 
+      visibility="public" static="false" abstract="false" return="int" name="size"
+      deprecated="not deprecated" final="true" bridge="true" native="false" synchronized="false" 
+      synthetic="false" 
+      >
+    </method>
+  </add-node>
+  <add-node
+    path="/api/package[@name='kotlin']/class[@name='UIntArray']"
+    >
+    <method 
+      visibility="public" static="false" abstract="false" return="int" name="size"
+      deprecated="not deprecated" final="true" bridge="true" native="false" synchronized="false" 
+      synthetic="false" 
+      >
+    </method>
+  </add-node>
+  <add-node
+    path="/api/package[@name='kotlin']/class[@name='UByteArray']"
+    >
+    <method 
+      visibility="public" static="false" abstract="false" return="int" name="size"
+      deprecated="not deprecated" final="true" bridge="true" native="false" synchronized="false" 
+      synthetic="false" 
+      >
+    </method>
+  </add-node>
+
+  <attr
+    path="/api/package[@name='kotlin']/class[@name='UByteArray']/method[@name='set-VurrAj0' and count(parameter)=3 and parameter[1][@type='byte[]'] and parameter[2][@type='int'] and parameter[3][@type='byte']]/parameter[3]"
+    name="type"
+    >
+    ubyte
+  </attr>
+
+  <attr
+    path="/api/package[@name='kotlin']/class[@name='UByteArray']/method[@name='get-w2LRezQ' and count(parameter)=2 and parameter[1][@type='byte[]'] and parameter[2][@type='int']]"
+    name="return"
+    >
+    ubyte
+  </attr>
+  <attr
+    path="/api/package[@name='kotlin']/class[@name='UIntArray']/method[@name='set-VXSXFK8' and count(parameter)=3 and parameter[1][@type='int[]'] and parameter[2][@type='int'] and parameter[3][@type='int']]/parameter[3]"
+    name="managedType"
```

Comment:
I realize you should always use `managedType` and are never supposed to use `type`, but I think in these cases we actually do want `type`.  There is some custom marshalling code that gets generated for Kotlin unsigned types, and I suspect it won't get triggered with `managedType`.  Thus the code may not work correctly at runtime.

Reply:
> I realize you should always use `managedType` and are never supposed to use `type`, but I think in these cases we actually do want `type`. There is some custom marshalling code that gets generated for Kotlin unsigned types, and I suspect it won't get triggered with `managedType`. Thus the code may not work correctly at runtime.

You know I try not to `remove-node` and to change native `type` and `return`.

OK. Trying only `type` (only for `parameter`)!!